### ISSUE TITLE
RZ10012 warning cannot be upgraded to error in .editorconfig

### ISSUE THREAD

**Issue by TonyPatterson** [dotnet/roslyn#55921]
### Describe the bug
When adding the line

```
dotnet_diagnostic.RZ10012.severity = error
```

to `.editorconfig`, I would expect the compiler to report the following to report the razor markup

```
<NonExistentComponent />
```

as an RZ10012 error, but it only reports it as a warning.

**Comment by sharwell** [comment]
@pranavkm When RZ10012 is reported, is it reported at a location within a source file? While **.globalconfig** (and older ruleset files) support configuring the severity for all diagnostics in the build output, **.editorconfig** only supports configuring the severity of diagnostics reported in source files located in a directory underneath that **.editorconfig** file on disk. There may be a bug in the application of severities, but I want to make sure this is intended to be a supported scenario before turning it over to the compiler team.

**Comment by pranavkm** [comment]
The file is part of the project - however it appears as an `AdditionalFIle``  in the source generator.

**Comment by sharwell** [comment]
Thanks! Assigning to @mavasani for further investigation.

**Comment by mavasani** [comment]
> The file is part of the project, however it appears as an AdditionalFile in the source generator. Would that make a difference?

Yes, absolutely. Severities in editorconfig files only apply to _source_ files. Additional file location is considered a non-source location and can only be configured at project level via global config files or /warnaserror:ID command line switch.

@gewarren Probably this warrants a mention at https://docs.microsoft.com/dotnet/fundamentals/code-analysis/configuration-files#editorconfig.
